libtock/alarm_timer.c

```diff
@@ -71,7 +74,9 @@ static void callback( __attribute__ ((unused)) int unused0,
     // has the alarm not expired yet? (distance from `now` has to be larger or
     // equal to distance from current clock value.
     if (alarm->dt > now - alarm->reference) {
-      alarm_internal_set(alarm->reference, alarm->dt);
+      if (alarm_internal_set(alarm->reference, alarm->dt) != RETURNCODE_SUCCESS)
+        putnstr("Err: failed to set Alarm\n", sizeof("Err: failed to set Alarm"));
+
```

Comment:
Please add curly braces to your if statements. We try to avoid [goto fail](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/) by construction.